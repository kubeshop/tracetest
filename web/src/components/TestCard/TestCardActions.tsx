import {Dropdown, Menu} from 'antd';
import * as S from './TestCard.styled';
import {useOnDeleteCallback} from './useOnDeleteCallback';

interface IProps {
  testId: string;

  onDelete(): void;
}

const TestCardActions: React.FC<IProps> = ({testId, onDelete}) => {
  const onClick = useOnDeleteCallback(onDelete);
  return (
    <Dropdown
      overlay={<Menu items={[{key: 'delete', label: <p data-cy="delete">Delete</p>, onClick}]} />}
      placement="bottomLeft"
      trigger={['click']}
    >
      <span data-cy={`test-actions-button-${testId}`} className="ant-dropdown-link" onClick={e => e.stopPropagation()}>
        <S.ActionButton />
      </span>
    </Dropdown>
  );
};

export default TestCardActions;
