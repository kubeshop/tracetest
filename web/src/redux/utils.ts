import {PromiseWithKnownReason} from '@reduxjs/toolkit/dist/query/core/buildMiddleware/types';
import webSocketGateway, {IListenerFunction} from 'gateways/WebSocket.gateway';

export type {IListenerFunction} from 'gateways/WebSocket.gateway';

interface IInitWebSocketSubscription {
  listener: IListenerFunction;
  resource: string;
  waitToCleanSubscription: Promise<void>;
  waitToInitSubscription: PromiseWithKnownReason<any, any>;
}

/**
 * Implements RTK Query logic to receive streaming updates for persistent queries.
 * It allows a query to establish a WebSocket subscription, and apply updates to
 * the cached data when the information is received from the server.
 * For more information please take a look: https://redux-toolkit.js.org/rtk-query/usage/streaming-updates
 */
export async function initWebSocketSubscription({
  listener,
  resource,
  waitToCleanSubscription,
  waitToInitSubscription,
}: IInitWebSocketSubscription) {
  try {
    await waitToInitSubscription;
    webSocketGateway.subscribe(resource, listener);
  } catch {
    // no-op in case `waitToCleanSubscription` resolves before `waitToInitSubscription`,
    // in which case `waitToInitSubscription` will throw
  }
  await waitToCleanSubscription;
  webSocketGateway.unsubscribe(resource, '1234');
}
