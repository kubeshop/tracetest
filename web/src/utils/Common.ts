import {capitalize} from 'lodash';
import Env from './Env';

export const escapeString = (str: string): string => {
  // eslint-disable-next-line no-control-regex
  return str.replace(/[\\"']/g, '\\$&').replace(/\u0000/g, '\\0');
};

export const isBoolean = (value: string): boolean => value === 'true' || value === 'false';

export const isJson = (str: string) => {
  try {
    JSON.parse(str);
  } catch (e) {
    return false;
  }

  return Number.isNaN(Number(str)) && !isBoolean(str) && true;
};

export const getObjectIncludesText = (object: unknown, text: string = ''): boolean => {
  if (!text.length) return false;

  const searchTextLower = text.toLowerCase();
  const stringSpan = JSON.stringify(object).toLowerCase();

  return stringSpan.includes(searchTextLower);
};

export const downloadFile = (data: string, fileName: string): Element => {
  const element = document.createElement('a');
  const file = new Blob([data]);
  element.href = URL.createObjectURL(file);
  element.download = fileName;
  document.body.appendChild(element);
  element.click();
  return element;
};

export function singularOrPlural(noun: string, quantity: number) {
  if (quantity === 1) return noun;
  return `${noun}s`;
}

export function ordinalSuffixOf(i: number) {
  const j = i % 10;
  const k = i % 100;
  if (j === 1 && k !== 11) {
    return `${i}st`;
  }
  if (j === 2 && k !== 12) {
    return `${i}nd`;
  }
  if (j === 3 && k !== 13) {
    return `${i}rd`;
  }
  return `${i}th`;
}

export function getTotalCountFromHeaders(meta: any) {
  return Number(meta?.response?.headers.get('x-total-count') || 0);
}

export const getServerBaseUrl = () => {
  const {host, protocol} = window.location;
  const prefix = Env.get('serverPathPrefix');

  return `${protocol}//${host}${prefix}`;
};

export const ToTitle = (str: string) => {
  return capitalize(str.replace(/\W/g, ' '));
};

export const getIsValidUrl = (url: string): boolean => {
  try {
    return !!getParsedURL(url);
  } catch (e) {
    return false;
  }
};

export const getParsedURL = (rawUrl: string): URL => {
  if (!!rawUrl && !rawUrl.startsWith('http')) {
    return new URL(`http://${rawUrl}`);
  }

  return new URL(rawUrl);
};

export const withLowPriority =
  <T extends (...args: any[]) => any>(fn: T): ((...args: Parameters<T>) => Promise<ReturnType<T>>) =>
  (...args: Parameters<T>): Promise<ReturnType<T>> =>
    new Promise(resolve => {
      setTimeout(() => {
        resolve(fn(...args));
      }, 0);
    });

export const toPercent = (value: number) => {
  return `${value * 100}%`;
};

export const abbreviateNumber = (num: number): string => {
  const symbols = ['', 'k', 'M', 'G', 'T', 'P', 'E'];

  // tier to determine suffix
  // eslint-disable-next-line no-bitwise
  const tier = (Math.log10(num) / 3) | 0;

  // if zero not need a suffix
  if (tier === 0) return num.toString();

  // get suffix
  const suffix = symbols[tier];
  if (!suffix) return num.toString();

  // get scale
  const scale = 10 ** (tier * 3);
  const scaled = num / scale;
  const rounded = scaled.toFixed(1);

  return rounded + suffix;
};
