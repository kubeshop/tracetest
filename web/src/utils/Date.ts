import {format, formatDistanceToNowStrict, isValid, parseISO} from 'date-fns';
import dropWhile from 'lodash/dropWhile';
import round from 'lodash/round';

export const ONE_MILLISECOND = 1000 * 1;
const ONE_SECOND = 1000 * ONE_MILLISECOND;
const ONE_MINUTE = 60 * ONE_SECOND;
const ONE_HOUR = 60 * ONE_MINUTE;
const ONE_DAY = 24 * ONE_HOUR;

const UNIT_STEPS: {unit: string; microseconds: number; ofPrevious: number}[] = [
  {unit: 'd', microseconds: ONE_DAY, ofPrevious: 24},
  {unit: 'h', microseconds: ONE_HOUR, ofPrevious: 60},
  {unit: 'm', microseconds: ONE_MINUTE, ofPrevious: 60},
  {unit: 's', microseconds: ONE_SECOND, ofPrevious: 1000},
  {unit: 'ms', microseconds: ONE_MILLISECOND, ofPrevious: 1000},
  {unit: 'Î¼s', microseconds: 1, ofPrevious: 1000},
];

const Date = {
  format(date: string, dateFormat = "EEEE, yyyy/MM/dd 'at' HH:mm:ss") {
    const isoDate = parseISO(date);
    if (!isValid(isoDate)) {
      return '';
    }
    return format(isoDate, dateFormat);
  },

  getTimeAgo(date: string) {
    const isoDate = parseISO(date);
    if (!isValid(isoDate)) {
      return '';
    }
    return formatDistanceToNowStrict(isoDate, {addSuffix: true});
  },

  isDefaultDate(date: string) {
    return date === '0001-01-01T00:00:00Z';
  },

  /**
   * Format duration for display.
   *
   * @param {number} duration - microseconds
   * @return {string} formatted duration
   */
  formatDuration(duration: number): string {
    // Drop all units that are too large except the last one
    const [primaryUnit, secondaryUnit] = dropWhile(
      UNIT_STEPS,
      ({microseconds}, index) => index < UNIT_STEPS.length - 1 && microseconds > duration
    );

    if (primaryUnit.ofPrevious === 1000) {
      // If the unit is decimal based, display as a decimal
      return `${round(duration / primaryUnit.microseconds, 2)}${primaryUnit.unit}`;
    }

    const primaryValue = Math.floor(duration / primaryUnit.microseconds);
    const primaryUnitString = `${primaryValue}${primaryUnit.unit}`;
    const secondaryValue = Math.round((duration / secondaryUnit.microseconds) % primaryUnit.ofPrevious);
    const secondaryUnitString = `${secondaryValue}${secondaryUnit.unit}`;
    return secondaryValue === 0 ? primaryUnitString : `${primaryUnitString} ${secondaryUnitString}`;
  },
};

export default Date;
