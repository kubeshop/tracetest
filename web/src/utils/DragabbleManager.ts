import _get from 'lodash/get';
import type React from 'react';

type TNil = null | undefined;

type TDraggableBounds = {
  clientXLeft: number;
  maxValue?: number;
  minValue?: number;
  width: number;
};

type TDraggableUpdate = {
  event: React.MouseEvent<HTMLDivElement | SVGSVGElement, MouseEvent> | MouseEvent;
  resetBounds(): void;
  value: number;
  x: number;
};

type TDraggableManagerOptions = {
  onGetBounds: () => TDraggableBounds;
  onDragEnd?: (update: TDraggableUpdate) => void;
  onDragMove?: (update: TDraggableUpdate) => void;
  onDragStart?: (update: TDraggableUpdate) => void;
};

interface IDraggableManager {
  initEventHandler(): (event: MouseEvent) => void;
  cleanup(): void;
  resetBounds(): void;
}

const LEFT_MOUSE_BUTTON = 0;

function DraggableManager({onGetBounds, ...rest}: TDraggableManagerOptions): IDraggableManager {
  let bounds: TDraggableBounds | TNil;
  let isDragging: boolean;
  let onDragStart: ((update: TDraggableUpdate) => void) | TNil;
  let onDragMove: ((update: TDraggableUpdate) => void) | TNil;
  let onDragEnd: ((update: TDraggableUpdate) => void) | TNil;

  const resetBounds = () => {
    bounds = undefined;
  };

  const getBounds = () => {
    if (!bounds) {
      bounds = onGetBounds();
    }
    return bounds;
  };

  const getPosition = (clientX: number) => {
    const {clientXLeft, maxValue, minValue, width} = getBounds();
    let x = clientX - clientXLeft;
    let value = x / width;
    if (minValue != null && value < minValue) {
      value = minValue;
      x = minValue * width;
    } else if (maxValue != null && value > maxValue) {
      value = maxValue;
      x = maxValue * width;
    }
    return {value, x};
  };

  const stopDragging = () => {
    window.removeEventListener('mousemove', handleEvent);
    window.removeEventListener('mouseup', handleEvent);
    const style = _get(document, 'body.style');
    if (style) {
      style.removeProperty('userSelect');
    }
    isDragging = false;
  };

  const handleEvent = (event: MouseEvent | React.MouseEvent<HTMLDivElement | SVGSVGElement, MouseEvent>) => {
    const {button, clientX, type: eventType} = event;
    let handler: ((update: TDraggableUpdate) => void) | TNil;

    if (eventType === 'mousedown') {
      if (isDragging || button !== LEFT_MOUSE_BUTTON) {
        return;
      }

      window.addEventListener('mousemove', handleEvent);
      window.addEventListener('mouseup', handleEvent);

      const style = _get(document, 'body.style');
      if (style) {
        style.userSelect = 'none';
      }
      isDragging = true;

      handler = onDragStart;
    } else if (eventType === 'mousemove') {
      if (!isDragging) {
        return;
      }

      handler = onDragMove;
    } else if (eventType === 'mouseup') {
      if (!isDragging) {
        return;
      }

      stopDragging();
      handler = onDragEnd;
    } else {
      throw new Error(`invalid event type: ${eventType}`);
    }

    if (!handler) {
      return;
    }

    const {value, x} = getPosition(clientX);

    handler({
      event,
      value,
      x,
      resetBounds,
    });
  };

  return {
    initEventHandler() {
      isDragging = false;
      bounds = undefined;
      window.addEventListener('resize', this.resetBounds);
      onDragStart = rest.onDragStart;
      onDragMove = rest.onDragMove;
      onDragEnd = rest.onDragEnd;

      return handleEvent;
    },

    cleanup() {
      if (isDragging) {
        stopDragging();
      }
      window.removeEventListener('resize', this.resetBounds);
      bounds = undefined;
      onDragStart = undefined;
      onDragMove = undefined;
      onDragEnd = undefined;
    },

    resetBounds,
  };
}

export default DraggableManager;
