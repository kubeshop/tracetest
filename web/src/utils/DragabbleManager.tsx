import _get from 'lodash/get';
import type React from 'react';

type TNil = null | undefined;

enum EUpdateTypes {
  DragEnd = 'DragEnd',
  DragMove = 'DragMove',
  DragStart = 'DragStart',
}

type TDraggableBounds = {
  clientXLeft: number;
  maxValue?: number;
  minValue?: number;
  width: number;
};

type TDraggableUpdate = {
  event: React.MouseEvent<HTMLDivElement | SVGSVGElement, MouseEvent> | MouseEvent;
  type: EUpdateTypes;
  value: number;
  x: number;
  resetBounds(): void;
};

type TDraggableManagerOptions = {
  getBounds: () => TDraggableBounds;
  onDragStart?: (update: TDraggableUpdate) => void;
  onDragMove?: (update: TDraggableUpdate) => void;
  onDragEnd?: (update: TDraggableUpdate) => void;
};

interface IDraggableManager {
  init(): (event: MouseEvent) => void;
  resetBounds(): void;
  dispose(): void;
}

const LEFT_MOUSE_BUTTON = 0;

function DraggableManager({getBounds, ...rest}: TDraggableManagerOptions): IDraggableManager {
  let _bounds: TDraggableBounds | TNil;
  let _isDragging: boolean;
  let _onDragStart: ((update: TDraggableUpdate) => void) | TNil;
  let _onDragMove: ((update: TDraggableUpdate) => void) | TNil;
  let _onDragEnd: ((update: TDraggableUpdate) => void) | TNil;

  const _resetBounds = () => {
    _bounds = undefined;
  };

  const _getBounds = () => {
    if (!_bounds) {
      _bounds = getBounds();
    }
    return _bounds;
  };

  const _getPosition = (clientX: number) => {
    const {clientXLeft, maxValue, minValue, width} = _getBounds();
    let x = clientX - clientXLeft;
    let value = x / width;
    if (minValue != null && value < minValue) {
      value = minValue;
      x = minValue * width;
    } else if (maxValue != null && value > maxValue) {
      value = maxValue;
      x = maxValue * width;
    }
    return {value, x};
  };

  const _stopDragging = () => {
    window.removeEventListener('mousemove', _handleDragEvent);
    window.removeEventListener('mouseup', _handleDragEvent);
    const style = _get(document, 'body.style');
    if (style) {
      style.removeProperty('userSelect');
    }
    _isDragging = false;
  };

  const _handleDragEvent = (event: MouseEvent | React.MouseEvent<HTMLDivElement | SVGSVGElement, MouseEvent>) => {
    const {button, clientX, type: eventType} = event;
    let type: EUpdateTypes | null = null;
    let handler: ((update: TDraggableUpdate) => void) | TNil;

    if (eventType === 'mousedown') {
      if (_isDragging || button !== LEFT_MOUSE_BUTTON) {
        return;
      }

      window.addEventListener('mousemove', _handleDragEvent);
      window.addEventListener('mouseup', _handleDragEvent);

      const style = _get(document, 'body.style');
      if (style) {
        style.userSelect = 'none';
      }
      _isDragging = true;

      type = EUpdateTypes.DragStart;
      handler = _onDragStart;
    } else if (eventType === 'mousemove') {
      if (!_isDragging) {
        return;
      }

      type = EUpdateTypes.DragMove;
      handler = _onDragMove;
    } else if (eventType === 'mouseup') {
      if (!_isDragging) {
        return;
      }

      _stopDragging();
      type = EUpdateTypes.DragEnd;
      handler = _onDragEnd;
    } else {
      throw new Error(`invalid event type: ${eventType}`);
    }

    if (!handler) {
      return;
    }

    const {value, x} = _getPosition(clientX);

    handler({
      event,
      type,
      value,
      x,
      resetBounds: _resetBounds,
    });
  };

  return {
    init() {
      _isDragging = false;
      _bounds = undefined;
      window.addEventListener('resize', this.resetBounds);
      _onDragStart = rest.onDragStart;
      _onDragMove = rest.onDragMove;
      _onDragEnd = rest.onDragEnd;

      return _handleDragEvent;
    },

    resetBounds: _resetBounds,

    dispose() {
      if (_isDragging) {
        _stopDragging();
      }
      window.removeEventListener('resize', this.resetBounds);
      _bounds = undefined;
      _onDragStart = undefined;
      _onDragMove = undefined;
      _onDragEnd = undefined;
    },
  };
}

export default DraggableManager;
