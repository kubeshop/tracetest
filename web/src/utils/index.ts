import {COMPARE_OPERATOR} from 'types';

const flattenAttributesSelector = () =>
  `resourceSpans[].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0]},resource.attributes[].{key:key,value: value.*|[0]}]|[][]`;

export const filterBySpanId = (spanId: string = '') =>
  `resourceSpans[?instrumentationLibrarySpans[?spans[?starts_with(spanId,'${spanId}')]]] | [].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0],type:'span'},resource.attributes[].{key:key,value: value.*|[0],type:'resource'}]|[][]`;

export const getOperator = (op: COMPARE_OPERATOR) => {
  switch (op) {
    case COMPARE_OPERATOR.EQUALS:
      return 'eq';
    case COMPARE_OPERATOR.NOTEQUALS:
      return 'ne';
    case COMPARE_OPERATOR.GREATERTHAN:
      return 'gt';
    case COMPARE_OPERATOR.LESSTHAN:
      return 'lt';
    case COMPARE_OPERATOR.GREATOREQUALS:
      return 'gte';
    case COMPARE_OPERATOR.LESSOREQUAL:
      return 'lte';
    default:
      return 'eq';
  }
};
