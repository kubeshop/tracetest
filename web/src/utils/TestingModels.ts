import {HTTP_METHOD} from '../constants/Common.constants';
import {CompareOperator} from '../constants/Operator.constants';
import {SemanticGroupNames} from '../constants/SemanticGroupNames.constants';
import {LOCATION_NAME} from '../constants/Span.constants';
import {SpanAttributeType} from '../constants/SpanAttribute.constants';
import {TestState} from '../constants/TestRunResult.constants';
import {IAssertion, IAssertionResult, IItemSelector, ISpanAssertionResult} from '../types/Assertion.types';
import {ISpan} from '../types/Span.types';
import {ITest} from '../types/Test.types';
import {ITestRunResult} from '../types/TestRunResult.types';
import {ITrace} from '../types/Trace.types';

interface IProps {
  mouseEvent: MouseEvent;
  assertionId: string;
  spanAssertionResult: ISpanAssertionResult;
  trace: ITrace;
  span: ISpan;
  test: ITest;
  assertion: IAssertion;
  assertionResult: IAssertionResult;
  testRunResult: ITestRunResult;
  resultId: string;
  testId: string;
  spanId: string;
}

const resultId = '0932u0e3';
const testId = '2309ei30';
const spanId = '4938u43';

const test: ITest = {
  assertions: [],
  description: '',
  lastTestResult: undefined,
  name: '',
  serviceUnderTest: {
    id: '',
    request: {
      auth: undefined,
      body: '',
      certificate: undefined,
      headers: undefined,
      method: HTTP_METHOD.GET,
      proxy: undefined,
      url: '',
    },
  },
  testId,
};

const span: ISpan = {
  attributeList: [],
  attributes: {},
  duration: 0,
  endTimeUnixNano: '',
  kind: '',
  name: '',
  parentSpanId: '',
  signature: [
    {
      locationName: LOCATION_NAME.SPAN,
      propertyName: '',
      value: '',
      valueType: '',
    },
  ],
  spanId,
  startTimeUnixNano: '',
  status: {code: ''},
  traceId: '',
  type: SemanticGroupNames.Http,
};

let selector: IItemSelector = {
  locationName: LOCATION_NAME.SPAN,
  propertyName: '',
  value: '',
  valueType: '',
};

const assertionResult: IAssertionResult = {
  assertion: {
    assertionId: '',
    selectors: [selector],
    spanAssertions: [],
  },
  spanListAssertionResult: [
    {
      resultList: [
        {
          actualValue: 'New request',
          comparisonValue: '',
          hasPassed: false,
          locationName: LOCATION_NAME.SPAN,
          operator: CompareOperator.EQUALS,
          propertyName: '',
          spanAssertionId: '',
          spanId: '',
          valueType: SpanAttributeType.intValue,
        },
      ],
      span,
    },
  ],
};

const testRunResult: ITestRunResult = {
  executionTime: 0,
  failedAssertionCount: 0,
  passedAssertionCount: 0,
  totalAssertionCount: 0,

  assertionResult: [{assertionId: '', spanAssertionResults: []}],
  assertionResultState: false,
  completedAt: '',
  createdAt: new Date().toISOString().toString(),
  response: undefined,
  resultId: '',
  spanId: '',
  state: TestState.CREATED,
  testId: '',
  trace: {
    description: '',
    resourceSpans: undefined,
    spans: [span],
  },
  traceId: '',
};

const spanAssertionResult: ISpanAssertionResult = {
  actualValue: 'New request',
  comparisonValue: '',
  hasPassed: false,
  locationName: LOCATION_NAME.INSTRUMENTATION_LIBRARY,
  operator: CompareOperator.EQUALS,
  propertyName: '',
  spanAssertionId: '',
  spanId: '',
  valueType: SpanAttributeType.doubleValue,
};

const assertionId = '234234';

const assertion: IAssertion = {
  assertionId,
  selectors: [
    {
      locationName: LOCATION_NAME.INSTRUMENTATION_LIBRARY,
      propertyName: '',
      value: '',
      valueType: '',
    },
  ],
  spanAssertions: [],
};
export const TestingModels: IProps = {
  mouseEvent: new MouseEvent('click', {
    bubbles: true,
    cancelable: true,
  }),
  assertionId,
  assertion,
  spanAssertionResult,
  assertionResult,
  test,
  testRunResult,
  span,
  resultId,
  testId,
  spanId,
  trace: {
    description: '',
    spans: [span],
  },
};

class NewTestingModel {
  test: ITest = test;
  assertionResult: IAssertionResult = assertionResult;
  testRunResult: ITestRunResult = testRunResult;
  span: ISpan = span;
  resultId: string = resultId;
  testId: string = testId;
  spanId: string = spanId;
  trace: ITrace = {
    description: '',
    spans: [span],
  };
}

export const TestingObj = new NewTestingModel();
