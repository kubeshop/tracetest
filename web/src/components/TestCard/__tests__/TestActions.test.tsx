import {render, waitFor} from '@testing-library/react';
import {TestingModels} from '../../../utils/TestingModels';
import TestCardActions from '../TestCardActions';

test('TestCardActions', async () => {
  const onDelete = jest.fn();
  const {getByTestId, container} = render(<TestCardActions onDelete={onDelete} testId={TestingModels.testId} />);
  await waitFor(() => getByTestId(`test-actions-button-${TestingModels.testId}`));
  expect(container).toMatchSnapshot();
  // fireEvent(getByTestId(`test-actions-button-${TestingModels.test.testId}`), TestingModels.mouseEvent);
});
