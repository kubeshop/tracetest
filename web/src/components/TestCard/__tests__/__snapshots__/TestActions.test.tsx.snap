// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestCardActions 1`] = `
<div>
  <span
    class="ant-dropdown-trigger ant-dropdown-link"
    data-cy="test-actions-button-2309ei30"
  >
    <span
      aria-label="more"
      class="anticon anticon-more sc-iCfMLu"
      role="img"
      style="font-size: 24px; color: rgb(154, 163, 171); cursor: pointer;"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</div>
`;
